### PR TITLE
[core] Style class property transitions are optional

### PR DESCRIPTION
**Pull request by brunoabinader**
We should use `optional<PropertyTransition>` following the trend on Map API.

/cc @jfirebaugh

### CODE REVIEW

File: src/mbgl/map/map_context.cpp [hunk 1]
```diff
@@ -297,15 +297,15 @@ bool MapContext::hasClass(const std::string& className) const {
...
 }
 
-void MapContext::removeClass(const std::string& className, const PropertyTransition& properties) {
+void MapContext::removeClass(const std::string& className, optional<PropertyTransition> properties) {
     if (style->removeClass(className, properties)) updateAsync(Update::Classes);
 }
 
-void MapContext::setClasses(const std::vector<std::string>& classNames, const PropertyTransition& properties) {
+void MapContext::setClasses(const std::vector<std::string>& classNames, optional<PropertyTransition> properties) {
     style->setClasses(classNames, properties);
```
Comment (tmpsantos) [inline src/mbgl/map/map_context.cpp]: The crash is because style could be `null` here.

Comment (brunoabinader) [inline src/mbgl/map/map_context.cpp]: This is not related to the style crash - this is an API cosmetics.

**Comment by jfirebaugh** [comment]
It looks to me like `PropertyTransition` is like `CameraOptions`: a set of parameters, each of which is individually `optional`. I'm sure that making the object as a whole `optional` is beneficial.

**Comment by jfirebaugh** [comment]
Edit: missed a "not".

**Comment by brunoabinader** [comment]
> It looks to me like PropertyTransition is like CameraOptions: a set of parameters, each of which is individually optional. I'm not sure that making the object as a whole optional is beneficial.

Agreed - this was mostly cosmetics to keep up to par with the other functions available in `Map`, but your argument on its members being `optional` already makes this irrelevant.
